src/language/typesystem/checker.ts

```diff
@@ -218,17 +217,53 @@ export class TypeChecker extends AstVisitor<Type, TypeScope> {
     return VOID_TYPE;
   }
 
-  /* istanbul ignore next */
-  public visitFunctionDeclaration(_: FunctionDeclaration, __: TypeScope): Type {
-    throw new Error('Not supported');
+  public visitFunctionDeclaration(
+    node: FunctionDeclaration,
+    scope: TypeScope
+  ): Type {
+    if (node.returnType === undefined) {
```

Comment:
nit: What do you think of making the default return type `void`?

It means you could write `main -> print('Hello')` instead of `main: void -> print('Hello')`

Reply:
That sounds reasonable